sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppImpl.java

```diff
@@ -209,16 +275,92 @@ Mono<Indexable> submitAppSettings() {
                     Mono
                         .fromCallable(
                             () -> {
-                                currentStorageAccount = storageAccountToSet;
-                                storageAccountToSet = null;
-                                storageAccountCreatable = null;
+                                resetStorageInfo();
                                 return this;
                             }));
         }
     }
 
+    private void resetStorageInfo() {
+        currentStorageAccount = storageAccountToSet;
+        storageAccountToSet = null;
+        storageAccountCreatable = null;
+    }
+
+    @Override
+    Mono<StringDictionaryInner> updateAppSettings(StringDictionaryInner appSettings) {
+        if (isFunctionAppOnACA()) {
+            return pollResult(
+                this.manager()
+                    .serviceClient()
+                    .getWebApps()
+                    .updateApplicationSettingsWithResponseAsync(resourceGroupName(), name(), appSettings),
+                StringDictionaryInner.class
+            );
+        } else {
+            return super.updateAppSettings(appSettings);
+        }
+    }
+
+    @Override
+    Mono<SiteConfigResourceInner> createOrUpdateSiteConfig(SiteConfigResourceInner siteConfig) {
+        if (isFunctionAppOnACA()) {
+            return pollResult(
+                this.manager()
+                    .serviceClient()
+                    .getWebApps()
+                    .createOrUpdateConfigurationWithResponseAsync(resourceGroupName(), name(), siteConfig),
+                SiteConfigResourceInner.class);
+        } else {
+            return super.createOrUpdateSiteConfig(siteConfig);
+        }
+    }
+
+    private <T> Mono<T> pollResult(Mono<Response<T>> responseMono, Class<T> responseBodyType) {
+        return responseMono.flatMap((Function<Response<T>, Mono<T>>) response -> {
+            if (response.getStatusCode() == 200) {
```

Comment:
Are there constants for the status codes somewhere and if not wouldn't it make sense to introduce them?

Reply:
Switched to use directive and this method is no longer needed.

Yeah, it's a valid point. Will consider introducing if encountered next time, thanks!